### PR TITLE
[ESIMD] Process every function in LowerESIMD

### PR DESCRIPTION
**Pull request by DenisBakhvalov**
This patch is a preparation for moving ESIMD-specific passes
into sycl-post-link. Right now, only functions marked with
"sycl-explicit-simd" metadata are lowered in the LowerESIMD pass.
In sycl-post-link, once we split ESIMD kernels from SYCL kernels,
there is no shared code between two types of kernels, and
LowerESIMD pass can safely process a module with ESIMD kernels
without worrying about another type of kernels (SYCL).

This change is also safe to do now since, as of today, we do not
allow mixing SYCL and ESIMD kernels in one source and in one program.
Thus, LowerESIMD can operate in "exclusive" mode without worrying
about SYCL kernels.

The ESIMD CFG markup made in `Sema::MarkSyclSimd` will be
removed in subsequent patches.

### CODE REVIEW

File: llvm/test/SYCLLowerIR/esimd_lower_intrins.ll [hunk 1]
```diff
@@ -14,7 +14,9 @@ target triple = "spir64-unknown-unknown-sycldevice"
 @vg = dso_local global %"cm::gen::simd<int, 16>" zeroinitializer, align 64 #0
 @vc = dso_local addrspace(1) global <32 x i32> zeroinitializer
 
-define dso_local spir_func <32 x i32>  @FUNC_1() !sycl_explicit_simd !1 {
+; LowerESIMD pass should process every function, 
+; !sycl_explicit_simd metadata is not necessary.
+define dso_local spir_func <32 x i32>  @FUNC_1() {
```
Comment (kbobrovs) [inline llvm/test/SYCLLowerIR/esimd_lower_intrins.ll:19]: There are a number of other functions with this metadata used for LowerESIMD testing: 
- right below in this test
- in other tests such as llvm/test/SYCLLowerIR/scalar_fptoui.ll

they should be corrected as well then, I think.

**Review by kbobrovs — CHANGES_REQUESTED** [review]

File: llvm/test/SYCLLowerIR/esimd_lower_intrins.ll (hunk 1 above)
Comment (DenisBakhvalov) [reply llvm/test/SYCLLowerIR/esimd_lower_intrins.ll:19]: Ok, will do.

Comment (DenisBakhvalov) [reply llvm/test/SYCLLowerIR/esimd_lower_intrins.ll:19]: @kbobrovs , done.

**Review by kbobrovs — APPROVED** [review]
Thanks. LGTM.

**Pull request merged**